### PR TITLE
Resolve file name.: custom: ${file(ymls/custom.yml)}

### PR DESCRIPTION
**Pull request by thiagoananias**
### Background
I'am having problems because i'am using a file reference path... this is just a way of solving... i don't know if it is the best way.

### CODE REVIEW

**Comment by fernando-mc** [comment]
Can you add a bit more information here? Sorry, I'm a little confused on what this is for? You're using a custom yaml file for configuration information and you want that to be supported by this?

**Comment by thiagoananias** [comment]
The configuration of my Serverless.yml are not all in the same file, the plugin does not work when i divide the configuration in files.

provider:
  name: aws
  runtime: nodejs8.10
  memorySize: 128
  stage: dev
  logRetentionInDays: 7
  region: ${self:custom.regions.${self:provider.stage}}
  versionFunctions: false
  timeout: 30
  stackTags:
    stage: ${self:provider.stage}
    region: ${self:provider.region}
    service: ${self:service}
  environment: ${file(ymls/environment.yml)}

custom: ${file(ymls/custom.yml)}
resources:
  Parameters: ${file(ymls/parameters.yml)}
  Conditions: ${file(ymls/conditions.yml)}
  Resources: ${file(ymls/resources.yml)}

**Comment by thiagoananias** [comment]
Before the serverless initiate i'm customizing the YML files with some other third party informations

**Comment by dirgapeter** [comment]
from https://serverless.com/framework/docs/providers/aws/guide/plugins/

> Variable references in the serverless instance are not resolved before a Plugin's constructor is called, so if you need these, make sure to wait to access those from your hooks.

I do suggest to put `this.options = this.serverless.service.custom.client;` inside `_validateConfig` and initialize with `{}` in constructor.

**Comment by jodanher** [comment]
> from https://serverless.com/framework/docs/providers/aws/guide/plugins/
> 
> > Variable references in the serverless instance are not resolved before a Plugin's constructor is called, so if you need these, make sure to wait to access those from your hooks.
> 
> I do suggest to put `this.options = this.serverless.service.custom.client;` inside `_validateConfig` and initialize with `{}` in constructor.

This is a better solution!

File: index.js [hunk 1]
```diff
@@ -122,6 +122,8 @@ class Client {
       routingRules,
       manageResources;
 
+	  this.options = this.serverless.service.custom.client;
```
Comment (Shereef) [inline index.js:125]: Please help us evaluate the issue and your fix by doing the following:

- Rebase off of master
- Format this file so this line is aligned well
- Test with serverless 3
- provide us with a repo that has the minimal code to reproduce this issue 

I realize this is a very old PR and may not be needed anymore so please help us validate that by doing the above steps

**Review by Shereef — CHANGES_REQUESTED** [review]

**Pull request closed without merging**